src/vue/modules/verification/general-form/store/index.js

```diff
@@ -0,0 +1,276 @@
+import { DOCUMENT_TYPES } from '@/js/const/document-types.const'
+import { ID_DOCUMENT_TYPES } from '../id-document-types'
+import { COUNTRIES } from '../countries'
+import { BLOB_TYPES } from '@tokend/js-sdk'
+import { types } from './types'
+import { api } from '../_api'
+
+import { wrapDocument } from './wrap_document'
+import { uploadDocument } from './upload_document'
+
+const state = {
+  isAccredited: false,
+  personal: {
+    firstName: '',
+    lastName: '',
+    dateOfBirth: '',
+  },
+  address: {
+    line1: '',
+    line2: '',
+    city: '',
+    state: '',
+    country: '', // { code, translation }
+    postalCode: '',
+  },
+  documents: {
+    idDocumentType: '', // { value, translationId }
+
+    avatar: null,
+    selfie: null,
+    idDocumentFace: null,
+    idDocumentBack: null,
+    proofOfInvestor: null,
+  },
+}
+
+const mutations = {
+  [types.SET_IS_ACCREDITED] (state, isAccredited) {
+    state.isAccredited = isAccredited
+  },
+
+  // personal
+  [types.SET_FIRST_NAME] (state, firstName) {
+    state.personal.firstName = firstName
+  },
+  [types.SET_LAST_NAME] (state, lastName) {
+    state.personal.lastName = lastName
+  },
+  [types.SET_DATE_OF_BIRTH] (state, dateOfBirth) {
+    state.personal.dateOfBirth = dateOfBirth
+  },
+
+  // address
+  [types.SET_LINE_1] (state, line1) {
+    state.address.line1 = line1
+  },
+  [types.SET_LINE_2] (state, line2) {
+    state.address.line2 = line2
+  },
+  [types.SET_CITY] (state, city) {
+    state.address.city = city
+  },
+  [types.SET_COUNTRY] (state, country) {
+    state.address.country = country
+  },
+  [types.SET_STATE] (state, s) {
+    state.address.state = s
+  },
+  [types.SET_POSTAL_CODE] (state, code) {
+    state.address.postalCode = code
+  },
+
+  // document fields
+  [types.SET_ID_DOCUMENT_TYPE] (state, type) {
+    state.documents.idDocumentType = type
+  },
+
+  // documents
+  [types.SET_AVATAR] (state, doc) {
+    state.documents.avatar = doc
+  },
+  [types.SET_SELFIE] (state, doc) {
+    state.documents.selfie = doc
+  },
+  [types.SET_ID_DOCUMENT_FACE] (state, doc) {
+    state.documents.idDocumentFace = doc
+  },
+  [types.SET_ID_DOCUMENT_BACK] (state, doc) {
+    state.documents.idDocumentBack = doc
+  },
+  [types.SET_PROOF_OF_INVESTOR] (state, doc) {
+    state.documents.proofOfInvestor = doc
+  },
+}
+
+const actions = {
+  async [types.GET_BLOB_DATA] (v, blobId) {
+    const { data } = await api().getWithSignature(`/blobs/${blobId}`)
+
+    return JSON.parse(data.value)
+  },
+
+  [types.POPULATE_STATE] ({ commit }, blobData) {
```

Comment:
@max-nes blob is an entity that already is stored in the API. it is the data it will store, not the blob itself. I don't like naming variables with the "data" word also, but here it means exactly what it means

Reply:
I mean the blob is the resource, data is it's attribute